LMS - Answer summaries and no-primary routing

bitdivision opened the pull request:
### What is the context of this PR?

This change includes:

- Duplicate the schema for LMS to allow for inclusion of the primary person.
  The new section has ids of `no-primary-...`.

- There is a fix for navigation to AnswerSummaries

- A fix for skip conditions to skip properly when the driving question has a group-instance-id and isn't in a repeating group.

- Add AnswerSummary to who lives here.

- Skip section in navigation if all groups are skipped or invalid (fixes bug with 'about-the-household')

- Content fix for LMS
### How to review 
Check that who lives here in LMS works as expected.

Known Issues:
- Navigation for who lives here will show 'About the household' as completed when it isn't

### Checklist

* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

bitdivision commented:
@dwyeradam - I've reverted the change to use `navigable_groups` for the moment. It was breaking the tests because interstitials were previously being shown in navigation, but weren't after the change.

I don't know if we rely on that functionality anywhere, but it was specifically included in the tests, so I assume interstitials were intended to show up in navigation.

MebinAbraham reviewed (commented):
This needs changing to match the original path:
https://github.com/ONSdigital/eq-survey-runner/blob/281342aa6fabb9e4138cf4b94a83c031e9f64e57/data/en/lms_2.json#L735

https://github.com/ONSdigital/eq-survey-runner/blob/281342aa6fabb9e4138cf4b94a83c031e9f64e57/data/en/lms_2.json#L811

MebinAbraham reviewed (dismissed):
This is the no-primary actual hours, but it references `two-jobs-j1-usual-hours-answer` instead of `no-primary-two-jobs-j1-usual-hours-answer`, which results in a 500 error.

https://github.com/ONSdigital/eq-survey-runner/blob/281342aa6fabb9e4138cf4b94a83c031e9f64e57/data/en/lms_2.json#L13910-L13924

Same thing for `two-jobs-j2-usual-hours-answer`.
Think there are quite a few routing rules which reference the wrong version for the id.

bitdivision committed:
LMS Final Changes

Duplicate schema to allow for inclusion of primary person
fix for navigation to answer summaries
fix for skip conditions when driving question has group-instance-id and isnt in repeating group
add answersummary to who lives here in LMS
skip section in navigation if all groups are skipped or invalid
Content fixes in LMS

MebinAbraham reviewed (commented):
Some routing seems to be invalid/missing.

The overtime questions are not being asked when:
- Non-primary
- 1 Job Working
- Some overtime

ajmaddaford reviewed (approved):
I'm happy that this works as expected for LMS. It's not very clear what the intentions are behind the checking of dependent groups/blocks/drivers and their relevance to getting answers. This is something we need to review in detail post LMS.

dwyeradam reviewed (approved):
Good enough for a Friday.
